garak/generators/litellm.py

```diff
@@ -142,6 +155,8 @@ def __init__(self, name: str = "", generations: int = 10, config_root=_config):
                         " or in the configuration file"
                     )
 
+        self._load_client()
```

Comment:
Provider check is still needed and the key extraction should have been moved into a custom `_validate_env_var()` implementation. This looks like something I missed in #711 that incorrectly enforces an API key as required for all `provider` values.
```suggestion
        self._load_client()

    def _validate_env_var(self):
       if self.provider is None:
            raise ValueError(
                "litellm generator needs to have a provider value configured - see docs"
            )
        elif (
            self.api_key is None
        ):
            # TODO: add other providers
            if self.provider == "openai":
                super()._validate_env_var()       
```

Reply:
So, uh, while dealing with this slow module, I started getting a failed test, but noticed that I had an openai env var key set, which meant the test actually ran. Have you ever seen the `litellm` tests pass? Looking at the tests we have, and the basic code examples on their website (see eg the invocations on https://docs.litellm.ai/docs/), the `provider` check seems to block intended functionality.

Reply:
I have never had them run and pass as they both required keys, I had noted in the original PR that it seemed like a config file would be required to instantiate the class. Although there was a comment that said it was not required, the original [embedded config parsing](https://github.com/leondz/garak/pull/572/files#diff-ff886897d971b8e468ed44457e63890380c3c95e187ee701c2c34e61699bcf44R106-R134) did *require* a provider. If `provider` was not found in `_config.plugins.generators["litellm.LiteLLMGenerator"]` it would raise a `ValueError`.

I intend to validate function as part of the testing here by setting up a local instance, however there is another issue with this class as the `torch_dtype` value cannot be accepted as a string.  I have fixes for this in progress in the refactor branch I am working one.  Short term I was intending to manually patch the `torch_dtype` default value to allow testing of this change in isolation.

Reply:
After some testing, I have validated that `LiteLLM` as implemented does required a `provider` from the config file.  This can be enforced using something like the suggestion updated in the top of this thread and would supply a method to suppress the api key requirement when supplying something other than `openai` as the `provider`.

A future PR can also expand the testing to provide a mock config that will would allow for mocking a response from the generator similar to the mock openai responses recently incorporated.

Reply:
I think the "as implemented" is something to be flexible on (see example in #755 ). This would remove the enforcement requirement. Unfortunately I don't have a good pattern for validating the input. I'm OK with relaxing the provider constraint and letting `litellm` do its own validation.